app/main/views/questionnaire.py

```diff
@@ -58,6 +58,9 @@ def post_questionnaire(eq_id, form_type, period_id, collection_id, location):
         return render_page(location, False)
 
     next_location = g.questionnaire_manager.get_current_location()
+    if next_location == location:
```

Comment:
If we created another end point for adding another composite field we wouldn't have to do this.

As discussed with @jonnyshaw89 it would also mean we could handle the logic explicitly there, therefore we wouldn't need to depend on the user action pattern which means we are in a better position for removing it